src/main/java/tc/oc/pgm/map/NameCacheUtil.java

```diff
@@ -0,0 +1,102 @@
+package tc.oc.pgm.map;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.UUID;
+import org.bukkit.Bukkit;
+import tc.oc.pgm.api.PGM;
+
+public class NameCacheUtil {
+
+  private static HashMap<UUID, String> cachedNames = new HashMap<UUID, String>();
+
+  public static boolean isUUIDCached(UUID uuid) {
+    return cachedNames.containsKey(uuid) && cachedNames.get(uuid) != null;
+  }
+
+  public static String getCachedName(UUID uuid) {
+    return cachedNames.get(uuid);
+  }
+
+  public static void addUUID(UUID uuid) {
+    if (!cachedNames.containsKey(uuid)) {
+      cachedNames.put(uuid, null);
+    }
+  }
+
+  private static String resolve(UUID uuid) throws IOException {
+    HttpURLConnection con =
+        (HttpURLConnection)
+            new URL("https://api.ashcon.app/mojang/v2/user/" + uuid.toString()).openConnection();
+    con.setRequestMethod("GET");
+    con.setRequestProperty("Accept", "application/json");
+    con.setConnectTimeout(5000);
+    con.setReadTimeout(5000);
+    BufferedReader br =
+        new BufferedReader(new InputStreamReader(con.getInputStream(), StandardCharsets.UTF_8));
+    StringBuilder response = new StringBuilder();
+    String line;
+    while ((line = br.readLine()) != null) {
+      response.append(line.trim());
+    }
+    br.close();
+    return new Gson().fromJson(response.toString(), JsonObject.class).get("username").getAsString();
```

Comment:
I'm pretty sure you can and should reuse the Gson object, declare it as a final static constant and reuse it elsewhere